tools/validate-changelog.sh

```diff
@@ -4,9 +4,23 @@
 # this should be sufficient to validate the CHANGELOG for our CI, provided that
 # every new tag has a corresponding CHANGELOG update, as we always parse the
 # CHANGELOG between two headers with a tagged version.
-if diff <(grep -ne '^# ' CHANGELOG.md) <(grep -ne '^# v[0-9]\+\.[0-9]\+\.[0-9]\+' CHANGELOG.md); then
-  echo "CHANGELOG validation PASSED!"
-else
+if ! diff <(grep -ne '^# ' CHANGELOG.md) <(grep -ne '^# v[0-9]\+\.[0-9]\+\.[0-9]\+' CHANGELOG.md); then
   echo "CHANGELOG validation FAILED! Headers must match the regex '^# v[0-9]\+\.[0-9]\+\.[0-9]\+.'"
   exit 1
 fi
+
+# check that all versions are unique
+versions=$(grep -o '^# v[0-9]\+\.[0-9]\+\.[0-9]\+' CHANGELOG.md)
+if [[ $(echo "${versions}" | sort | uniq -d | wc -l) -gt 0 ]]; then
+  echo "CHANGELOG validation FAILED! Duplicate versions found."
+  exit 1
+fi
```

Comment:
Let's print the dupes so they're easier to locate:
```suggestion
versions=$(grep -o '^# v[0-9]\+\.[0-9]\+\.[0-9]\+' CHANGELOG.md)
duplicates=$(echo "${versions}" | sort | uniq -d)
if [[ -n "${duplicates}" ]]; then
  echo "CHANGELOG validation FAILED! Duplicate versions found:"
  echo "${duplicates}"
  exit 1
fi
```